static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
 
 Record changes to the repository by saving outputs to cache.
 
+## Synopsis
+
 ```usage
     usage: dvc commit [-h] [-q | -v] [-f] [-d] [-R] [targets [targets ...]]
 
     positional arguments:
       targets               DVC files.
-
-    optional arguments:
-      -f, --force           Commit even if checksums for dependencies/outputs
-                            changed.
-      -d, --with-deps       Commit all dependencies of the specified target.
-      -R, --recursive       Commit cache for subdirectories of the specified
-                            directory.
 ```
 
+## Description
+
+Normally DVC commands like `dvc add`, `dvc repro` or `dvc run`, commit the data
+to the DVC cache as the last step. What _commit_ means is that DVC
+
+* Computes a checksum for the file
+* Enters the checksum and file name into the DVC stage file
+* Tells the SCM to ignore the file (e.g. add entry to `.gitignore`).  If the
+  workspace was initialized with no SCM support (`dvc init --no-scm`) this does
+  not happen.
+* Adds the file to the DVC cache
+
+There are many cases where the last step is not desirable. For the DVC commands
+where it is appropriate the `--no-commit` option prevents the last step from
+occurring. The checksum is still computed and added to the DVC file, but the
+file is not added to the cache.
+
+That's where the `dvc commit` command comes into play. It handles that last
+step of adding the file to the DVC cache.
+
+The `dvc commit` command is useful for several scenarios where a stage is in
+development, or takes a long time to run, or must be run on another machine
+with higher compute power. The key is to run DVC commands in a mode where data
+is not immediately committed to the cache (the `--no-commit` or `--no-exec`
+options), and to commit the data as a separate step performed when it is certain
+the data is finalized.
+
+* Code for a stage is under active development, with lots of temporary files
+  sitting in the workspace. Committing those temporary files would clutter up
+  the cache, so it's better to commit them when development is finished.
+* Execution of a stage takes a long time to run, like 24 hours, or requires
+  specialized hardware like GPU's or an HPC cluster. The pipeline and cache
+  might be on one machine, like a laptop, and execution of the long-running
+  stage might be manually coordinated with the data added to the cache using
+  `dvc commit` when ready.
```

Comment:
Can we also mention that even though you can run `dvc commit` whenever you want and will save all the data and references in the whole pipeline as you wish, this is not the recommended way - this can break the reproducibility.

Reply:
Yes, have mentioned it